tapiriik/services/Pulsstory/pulsstory.py

```diff
@@ -189,57 +184,45 @@ def DownloadActivity(self, serviceRecord, activity):
         return activity
     
     def _convertList(self, rawData, listName):
-        streamData = []
         timeListName = listName + "Time"
         valueListName = listName + "Value"
-        try:
-            check = timeListName is not None and timeListName in rawData
-            check = check and valueListName is not None and valueListName in rawData
-            if check:
-                timeList = rawData[timeListName]
-                valueList = rawData[valueListName]            
-                Nt = len(timeList)
-                Nv = len(valueList)            
-                if Nt == Nv:
-                    for n in range(Nt):
-                        streamData.append((timeList[n], valueList[n]))
-                else:
-                    raise Exception("Different size of lists:" + listName);
-        except Exception as e:
-            pass
-            # logger.warn("Cannot convert list.", e)
-                    
-
-        return streamData
+        check = timeListName is not None and timeListName in rawData
+        check = check and valueListName is not None and valueListName in rawData
+        if check:
+            timeList = rawData[timeListName]
+            valueList = rawData[valueListName]
+            if isinstance(timeList, collections.Iterable) and isinstance(valueList, collections.Iterable):
+                result = list(zip(timeList, valueList))                                
+                return result
+
+        return []
        
     def _convertPathList(self, rawData):
         streamData = []
         timeListName = "PathTime"
         longitudeListName = "LongitudePathValue"
         latitudeListName = "LatitudePathValue"
-        try:   
-            check = timeListName is not None and timeListName in rawData
-            check = check and longitudeListName is not None and longitudeListName in rawData
-            check = check and latitudeListName is not None and latitudeListName in rawData
-            if check:
-                timeList = rawData[timeListName]
-                longitudeList = rawData[longitudeListName]
-                latitudeList = rawData[latitudeListName]                        
+        
+        check = timeListName is not None and timeListName in rawData
+        check = check and longitudeListName is not None and longitudeListName in rawData
+        check = check and latitudeListName is not None and latitudeListName in rawData
+        if check:
+            timeList = rawData[timeListName]
+            longitudeList = rawData[longitudeListName]
+            latitudeList = rawData[latitudeListName]
+            if isinstance(timeList, collections.Iterable) and isinstance(longitudeList, collections.Iterable) and isinstance(latitudeList, collections.Iterable):                        
```

Comment:
This is kind of an ugly check for iterability - what exactly is the failure case this anticipates? Maybe a comment or something, too.